web/src/components/MemoContent/Link.tsx

```diff
@@ -1,18 +1,96 @@
+import { Tooltip, Card, AspectRatio, Box } from "@mui/joy";
+import { Link as MLink } from "@mui/joy";
+import { useEffect, useState } from "react";
+import { metadataServiceClient } from "@/grpcweb";
+
 interface Props {
   url: string;
   text?: string;
 }
 
 const Link: React.FC<Props> = ({ text, url }: Props) => {
+  const [title, setTitle] = useState<string>();
+  const [description, setDescription] = useState<string>();
+  const [image, setImage] = useState<string>();
```

Comment:
Let's use the generated `LinkMetadata` type instead of split fields. 

```ts
const [linkMetadata, setLinkMetadata] = useState<LinkMetadata>();
```

Reply:
I have `Metadata` type and not `LinkMetadata` is that ok?

Reply:
I tweak its definition with https://github.com/usememos/memos/commit/5d740c3813b665ee5a6a1991e74eeab37343f6b7